continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerRunner.java

```diff
@@ -32,38 +32,29 @@
     private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss");
 
     @Override
-    @SuppressWarnings("BusyWait")
     public void run() {
         boolean started = false;
         String params = null;
-        while (!Thread.interrupted()) {
-            try {
-                if (stopFileExists()) {
-                    log.info("Stop file exists on filesystem: {}, will not run profiler and go to sleep for 10 minutes", properties.getStopFile());
-                    Thread.sleep(SleepTime.TEN_MINUTES);
-                } else {
-                    log.info("Starting async-profiler");
-                    params = createParams();
-                    asyncProfiler.execute("start," + params);
-                    started = true;
-                    Thread.sleep(properties.dumpIntervalMilliseconds());
-                    log.info("Stopping async-profiler");
+        try {
+            if (stopFileExists()) {
+                log.info("Stop file exists on filesystem: {}, will not run profiler", properties.getStopFile());
+            } else {
+                log.info("Starting async-profiler");
+                params = createParams();
+                asyncProfiler.execute("start," + params);
+                started = true;
+                log.info("Stopping async-profiler");
+                asyncProfiler.execute("stop," + params);
```

Comment:
it seems we start and immediately stop profiling. I think we should do it otherwise first stop current profiling (except the first time on start) then immediately start another profiling and wait for following execution of `ContinuousAsyncProfilerRunner`

Reply:
yes, that was stupid Thread.sleep removal